### ISSUE TITLE
semi-colon after a formula causes error

### ISSUE THREAD

**Issue by ajmay81** [doxygen/doxygen#10878]
As requested https://github.com/doxygen/doxygen/issues/10552#issuecomment-2120461400 is reported here as it's own issue. Here is an example `test.f90`:
```
!> test doxygen
!>
!> A formula \f$A\f$; followed by a semi-colon
subroutine test
end subroutine test
```
which generates:
```
/host> doxygen Doxyfile 1>/dev/null
********************************************************************
Error in file /host/test.f90 line: 6, state: 10(ModuleBody)
********************************************************************
/host> doxygen --version
1.11.0
```
There is no such problem if the semi-colon is changed to a comma.

**Comment by albert-github** [comment]
I've just pushed a proposed patch, pull request #10880

**Comment by ajmay81** [comment]
Thanks very much. For now I can just add a space before the semi-colon to avoid the problem and use 1.11.0 release, but will revert back when the fix is released in an official version.

**Comment by albert-github** [comment]
Code has been integrated in master on GitHub (please don't close the issue as this will be done at the moment of an official release).

**Comment by doxygen** [comment]
This issue was previously marked 'fixed but not released',
which means it should be fixed in doxygen version 1.12.0.
Please verify if this is indeed the case. Reopen the
issue if you think it is not fixed and please include any additional information
that you think can be relevant (preferably in the form of a self-contained example).
